src/Client/CurlHandler.php

```diff
@@ -57,8 +57,25 @@ public function __destruct()
         }
     }
 
+    /**
+     * @param array $request
+     *
+     * @return CompletedFutureArray
+     */
     public function __invoke(array $request)
     {
+        return new CompletedFutureArray(
+            $this->invokeAsArray($request)
+        );
+    }
+
+    /**
+     * @param array $request
+     *
+     * @return array
+     */
+    public function invokeAsArray(array $request)
```

Comment:
Some options are
- that this function __invoke always returns the array and that some decorator adds the CompleteFutureArray around it
- We dont decorate and the caller adds it himself 
- Allow the construct of CompleteFutureArray to receive itself

None sound realy ideal